### PR TITLE
Fix evaluation issues

### PR DESCRIPTION
**Pull request by Tixxx**
**Description**: 
Fix evaluation issues

**Motivation and Context**
Override drop ratio in eval with 0 so that Dropout behaves as an Identity node so that evaluation results are correct.
Added unit tests for python frontend.

### CODE REVIEW

**Comment by SherlockNoMad** [comment]
Overall, I find this approach a bit too intrusive to the framework. It's injecting a kernel level control from the RunOption. I think this is breaking the original design principle that an op is self-descriptive and stateless. Input+Attribute alone should determine the behavior of an op. 

If this flag is introduced, the op's behavior would depend on the config set by run time. 

For dropout, we have a "ratio" input to determine if it should invoke training mode. 
Similarly for BatchNorm-12, we introduced another input "is_training_mode". 

To make these nodes to operate in the eval mode, we can override the inputs of there nodes. Initializer of a node can still be override with graph feeds.

**Comment by jessebenson** [comment]
I agree with Sherlock, and he summarized the design quite well.

There's a relatively small number of operators that need different behaviors in training vs inferencing/evaluation mode.  We should favor controlling those behaviors with operator inputs, and if we need to control it per `session.run()`, then we can add that as a graph input.  As Sherlock said, we can have an initializer (for the default value) and still add it as a graph input so we can override it with the input feeds.

File: orttraining/orttraining/core/session/training_session.cc [hunk 1]
```diff
@@ -676,6 +679,28 @@ bool TrainingSession::IsGraphOutputFp32Node(const std::string& output_name) cons
...
+
+  // add ratio node to graph input for overriding.
+  GraphAugmenter::GraphDefs defs{};
+
+  for (const auto& node : graph.Nodes()) {
+    auto it = Dropout_Nodes.find(node.OpType());
+    if(it != Dropout_Nodes.cend()) {
+      auto& ratio_name = node.InputDefs()[1]->Name();
+      dropout_eval_feeds_.insert(ratio_name);
+      defs.AddGraphInputs({ratio_name});
```
[the remark this answers is not in the document]
Comment (Tixxx) [reply orttraining/orttraining/core/session/training_session.cc:728]: good catch.

File: orttraining/orttraining/python/ort_trainer.py [hunk 2]
```diff
@@ -85,12 +85,19 @@ def get_group_accumulated_gradients_output_node_arg_name(session):
...
     return accumulated_gradients_output_node_args[0].name
 
-def ort_training_session_run_helper(session, iobinding, inputs, input_descs, output_descs, device, run_options=None):
+def ort_training_session_run_helper(session, iobinding, inputs, input_descs, output_descs, device, run_options=None, additional_feeds=None):
     for input, input_desc in zip(inputs, input_descs):
         device_index = input_get_device_index(input)
         iobinding.bind_input(input_desc.name_, input.device.type, device_index, dtype_torch_to_numpy(input.dtype),
                              list(input.size()), input.data_ptr())
 
+    if additional_feeds:
```
[the remark this answers is not in the document]
Comment (Tixxx) [reply orttraining/orttraining/python/ort_trainer.py]: moved the logic to training session.

**Comment by liqunfu** [comment]
Do not forget to rebase and target to the master branch
